File: Day33/first_duplicate.py
"""Function to find the First duplicate in a array"""
def find_duplicate(arr):

    """Create an empty set to store unique elements"""
    seen = set()
    for num in arr:
        #check if the element is alredy in the set
        if num in seen:
            #If it is, return the element as the first duplicate
            return num
        #If not, add the element to the set
        seen.add(num)
    return None

input_arr= [2,1,3,5,3,2]
print(find_duplicate(input_arr))

File: Day33/firstDuplicate.js
//Function to find the First duplicate in a array
function getFirstDuplicate(arr){

    //Iterate over each element in the array
    for(let i = 0; i<arr.length; i++){

        //Compare the urrent element with every other element that comes after it
        for(let j = i+1; j<arr.length; j++){

            //Check if the elements are equal
            if(arr[i] == arr[j]){
                //If it is, return the element as the first duplicate
                return arr[i];
            }
        }
    }
    return null; // Return -1 if no duplicate are found 
}
let arr = [2,1,3,5,3,2];
console.log("First duplicate : " , getFirstDuplicate(arr));
